Update parameter-best-practices.md about secure string (#23302)
Azure/azure-powershell: documentation/development-docs/design-guidelines/parameter-best-practices.md

#### Secret Parameters

> For parameters whose type is string and which represent a value that should be kept secret in some fashion (such as a password, secret, key, etc.), the type of the parameter should be `SecureString` to limit the exposure of sensitive string data from unexpected leakage during cmdlet execution. The practice also applies to output properties whose type is string and that should be kept in secret.

Reviewer: Please link "SecureString" to .net ref doc.

> For parameters whose type is string and which represent a value that should be kept secret in some fashion (such as a password, secret, key, etc.), the type of the parameter should be `SecureString` to limit the exposure of sensitive string data from unexpected leakage during cmdlet execution. The practice also applies to output properties whose type is string and that should be kept in secret.

Author: ```suggestion
For parameters whose type is string and which represent a value that should be kept secret in some fashion (such as a password, secret, key, etc.), the type of the parameter should be [SecureString](https://learn.microsoft.com/dotnet/api/system.security.securestring) to limit the exposure of sensitive string data from unexpected leakage during cmdlet execution. The practice also applies to output properties whose type is string and that should be kept in secret.
```

Please notice that DO NOT use `SecureString` for encryption purposes. We only recommend to use `SecureString` as a wrapper of string to prevent unexpected leakage of information as string may still be exposed to any process or operation that has access to raw memory. 

> From [see SecureString shouldn't be used](https://github.com/dotnet/platform-compat/blob/master/docs/DE0001.md)

Author: ```suggestion
From [How secure is SecureString?](https://learn.microsoft.com/dotnet/api/system.security.securestring#how-secure-is-securestring)
```

#### Secret Parameters

> For parameters whose type is plain text and which represent a value that should be kept secret in some fashion (such as a password, secret, key, etc.), the type of the parameter should be `SecureString` to avoid any sensitive information about the parameter from leaking during cmdlet execution.

Reviewer: ```suggestion
For parameters whose type is plain text and which represent a value that should be kept secret in some fashion (such as a password, secret, key, etc.), the type of the parameter should be `SecureString` to avoid any sensitive information about the parameter from leaking unexpectly during cmdlet execution.
```

> Please notice that `SecureString` is not used for any encryption purposes. It is a wrapper of the plain text to prevent unexpected leakage of information. 

Reviewer: We can reference the words from .NET doc directly.
